src/storage-client/src/metrics.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Metrics for the storage controller components
+
+use std::sync::Arc;
+
+use mz_ore::cast::{CastFrom, TryCastFrom};
+use mz_ore::metric;
+use mz_ore::metrics::DeleteOnDropHistogram;
+use mz_ore::metrics::HistogramVecExt;
+use mz_ore::metrics::MetricsRegistry;
+use mz_service::codec::StatsCollector;
+
+use crate::client::{ProtoStorageCommand, ProtoStorageResponse};
+use crate::types::instances::StorageInstanceId;
+
+/// Compute controller metrics
+#[derive(Debug, Clone)]
+pub struct StorageControllerMetrics {
+    messages_sent_bytes: prometheus::HistogramVec,
+    messages_received_bytes: prometheus::HistogramVec,
+}
+
+impl StorageControllerMetrics {
+    pub fn new(metrics_registry: MetricsRegistry) -> Self {
+        Self {
+            messages_sent_bytes: metrics_registry.register(metric!(
+                name: "mz_storage_messages_sent_bytes",
+                help: "size of storage messages sent",
+                var_labels: ["host"]
```

Comment:
With the cluster unification, the concept of a storage host has been removed (you'll see when you rebase). I think we can instead use the instance/replica concept here, like for the compute metrics.